src/main/java/frc/team2412/robot/Subsystems/IndexerSubsystem.java

```diff
@@ -0,0 +1,304 @@
+	public DoubleSolenoid rearClutch;
+
+	public int numBalls = 0;
+	public boolean ballUnbalancedSide = false;
+
+	public IndexerSubsystem() {
+		// Motors
+		indexBackMotor = RobotMap.indexFrontMotor;
+		indexFrontMotor = RobotMap.indexBackMotor;
+		indexMidMotor = RobotMap.indexMidMotor;
+		indexMotors = new SpeedControllerGroup(indexFrontMotor, indexBackMotor, indexMidMotor);
+		indexSideMotors = new SpeedControllerGroup(indexFrontMotor, indexBackMotor);
+
+		// Sensors
+		back = RobotMap.BB;
+		backMid = RobotMap.BT;
+		mid = RobotMap.T;
+		frontMid = RobotMap.FT;
+		front = RobotMap.FF;
+		
+		//soulenoid
+		frontClutch = RobotMap.frontClutch;
+		rearClutch = RobotMap.rearClutch;
+	}
+
+	public void indexerLogic(boolean shoot) {
+		if (shoot) {
+			shoot();
+		} else {
+			switch (numBalls) {
+			case 0:
+				intake1();
+			case 1:
+				intake2();
+				switchBalls2(ballUnbalancedSide);
+			case 2:
+				intake3();
+				switchBalls34(ballUnbalancedSide);
+			case 3:
+				intake4();
+				switchBalls34(ballUnbalancedSide);
+			case 4:
+				intake5();
+			case 5:
+				System.out.println("im full");
+			}
+		}
+
+	}
+	//shifters
+	public void frontShiftUp() {
+		frontClutch.set(DoubleSolenoid.Value.kForward);
+	}
+	public void frontShiftDown() {
+		frontClutch.set(DoubleSolenoid.Value.kReverse);
+	}
+	public void backShiftUp() {
+		rearClutch.set(DoubleSolenoid.Value.kForward);
+	}
+	public void backShiftDown() {
+		rearClutch.set(DoubleSolenoid.Value.kReverse);
+	}
+	public void allShiftUp() {
+		rearClutch.set(DoubleSolenoid.Value.kForward);
+		frontClutch.set(DoubleSolenoid.Value.kForward);
+	}
+	public void allShiftDown() {
+		rearClutch.set(DoubleSolenoid.Value.kReverse);
+		frontClutch.set(DoubleSolenoid.Value.kReverse);
+	}
+	// Motors
+	public void stopAll() {
+		indexMotors.set(0.0);
+	}
+
+	public void stopSides() {
+		indexSideMotors.set(0.0);
+	}
+
+	// Sensors
+	public boolean allSensorsOn() {
+		if (back.get() && backMid.get() && mid.get() && frontMid.get() && front.get()) {
+			return true;
+		}
+
+		return false;
+		// check if all sensors are on
+	}
+
+	public boolean allSensorsOff() {
+		if (!back.get() && !backMid.get() && !mid.get() && !frontMid.get() && !front.get()) {
+			return true;
+		}
+
+		return false;
+		// check if all sensors are on
+	}
+
+	/*
+	 * if(allSensorsOff()){intake12();
+	 * 
+	 * boolean b;if(backMid.get()) { b = true; } else { b = false; }
+	 * switchBalls2(b); intake3(); if (backMid.get()) { b = true; } else { b =
+	 * false; } switchBalls34(b); intake4(); if (back.get()) { b = true; } else { b
+	 * = false; } switchBalls34(b); intake5(); }
+	 * 
+	 * shoot();
+	 * 
+	 * }
+	 */
+	public void shoot() {
+		allShiftDown();
+		indexMidMotor.set(1);
+		if (frontMid.get() && front.get()) {
+			indexFrontMotor.set(0);
+		} else if (backMid.get() && back.get()) {
+			indexBackMotor.set(0);
+		}
+		
+		stopAll();
+	}
+
+//Switch ball positions
+	public void switchBalls2(boolean currentDirection) {

```

Comment:
This (and most of the methods in this class) aren't used externally and don't appear to be ever used externally from the Indexer Commands.  Please change them to `private` to avoid external code unintentionally starting to use the internal implementation detail methods.